gui/src/app/FileEditor/TextEditor.tsx

```diff
@@ -60,10 +71,35 @@ const TextEditor: FunctionComponent<Props> = ({defaultText, text, onSaveText, ed
         if (editor.getValue() === editedText) return
         editor.setValue(editedText || defaultText || '')
     }, [editedText, editor, defaultText])
+    const [monacoInstance, setMonacoInstance] = useState<Monaco | undefined>(undefined)
+    useEffect(() => {
+        if (!monacoInstance) return
+        if (codeMarkers === undefined) return
+        if (editor === undefined) return
+        const model = editor.getModel()
+        if (model === null) return
+        const toSeverity = (s: 'error' | 'warning' | 'hint' | 'info'): monaco.MarkerSeverity => {
+            switch (s) {
+                case 'error': return monaco.MarkerSeverity.Error
+                case 'warning': return monaco.MarkerSeverity.Warning
+                case 'hint': return monaco.MarkerSeverity.Hint
+                case 'info': return monaco.MarkerSeverity.Info
+            }
+        }
```

Comment:
Mind moving this helper function out of the hook?

Reply:
Done